### ISSUE TITLE
configcontroller spams on new build 

### ISSUE THREAD

**Issue by PI-Victor** [openshift/origin#8494]
```
E0413 13:51:39.085548    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.092361    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.118223    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.147990    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.168136    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.222123    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.236496    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.271908    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.309329    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.324661    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.331913    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.340998    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
E0413 13:51:39.568130    4413 config_controller.go:82] error instantiating Build from BuildConfig demo/ruby-hello-world: imagestreamtags "ruby-22-centos7:latest" not found
I0413 13:51:40.355530    4413 trace.go:57] Trace "Create /oapi/v1/namespaces/demo/imagestreamimports" (started 2016-04-13 13:51:39.046866161 +0200 CEST):
```

Whenever kicking off a new build without the image being available locally the the config controller retries a lot of times. Not sure if it's docker replying back that the image is available or what. Maybe it needs some sort of delay or something.
@bparees @mfojtik @rhcarvalho  what do you think?

**Comment by mfojtik** [comment]
@bparees seems like the retry manager for config change trigger needs more intelligence... 

@PI-Victor is this default log level?

**Comment by bparees** [comment]
it's basically working as designed, we retry very fast until the token bucket is empty.  we can debate the log level, but trying to divide errors into "retry quickly" and "retry slowly" is going to get ugly.

the logging is handled by k8s' "handleError" logic though.

**Comment by bparees** [comment]
also you won't get that spam on every new build, only new builds where the imagestreamtag doesn't exist yet.  @PI-Victor can you confirm that?

**Comment by PI-Victor** [comment]
@bparees  yes. only happens when the imagestream isn't available yet and also it displays on loglevel 0. so if it's not serious i think it's best if we don't display this on log level 0, as @mfojtik suggested we can make it V(2). My question is now, can something else happen that we need to display in loglevel 0 that we would mute if we change the logic to log-level 2?

**Comment by bparees** [comment]
@PI-Victor we can't easily control when it gets logged because it goes through the k8s error handler, we aren't the ones directly logging it.  It might be worth investigating if it should be going through the k8s error handler, but that was the pattern we used.

As for distinguishing...  right now, not easily.  If you look at the code in question, we called Instantiate and got an error.  Right now we don't have any good way to differentiate what sort of error we got.  That's not to say we couldn't add that.  But we also have to keep in mind that at some point the build will fail if it retries long enough, so we need to make sure the final failure is logged.  

So fixing this definitely needs some careful thought.
